### PR TITLE
WT-8193 Wrong corner case in VLCS rollback-to-stable

### PR DESCRIPTION
**Pull request by sauclovian-wt**
Fix a corner case of VLCS rollback-to-stable.

It turns out that just the presence of a stable update for one key in the cell isn't sufficient to skip the whole cell. While it's true that in a fully timestamped world, if there's a stable update it must be newer than the cell's on-disk value and therefore the cell's on-disk value must be stable, this breaks down if the stable update isn't timestamped. We can skip iterating the cell if _every_ key in it has a stable update; otherwise we need to check each key.

It's also incorrect to call downstream to __rollback_abort_ondisk_kv for keys that have a stable update; the downstream code assumes that no stable update exists and will in some cases (e.g. in-memory databases) do the wrong thing if one does.

Therefore:
   - have __rollback_abort_insert_list return a count of keys with stable updates instead of just a flag;
   - add a __rollback_has_stable_update to report if an already processed update chain still has a valid and non-aborted (therefore stable) update on it;
   - rework the comment about this logic;
   - skip iterating the cell only as described above (or if the cell is deleted, as before);
   - when iterating the keys in the cell, also iterate the cell's update list to check for stable updates;
   - add a test that exhibits the failing case.

Note that because there's no way to prevent a page being reconciled before you're ready, without the changes the test might still occasionally pass. (Reconciling the page breaks the RLE cell in which things are supposed to happen and this makes the failure inaccessible.)

This test fails reliably for me without the fixes, but a more elaborate one I also wrote that tests a lot of related scenarios does
not. (So I'm not including that one.)

### CODE REVIEW

**Comment by lukech** [comment]
autobuild: Can one of the admins verify this patch?

**Comment by sauclovian-wt** [comment]
Hmm, that's odd, I pasted the description text from the jira ticket into the PR title, or I thought I did, but it seems to have disappeared.

File: test/suite/test_rollback_to_stable27.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        self.assertEqual(v, check_value)
+        self.assertEqual(evict_cursor.reset(), 0)
+        self.session.rollback_transaction()
+        evict_cursor.close()
+
+    def test_rollback_to_stable(self):
+        nrows = 10
+
+        # Create a table without logging.
+        uri = "table:rollback_to_stable28"
```
Comment (kommiharibabu) [inline test/suite/test_rollback_to_stable27.py:74]: The uri name also can be adjusted according to the test name.

**Review by kommiharibabu — APPROVED** [review]
LGTM. A small change in the test is required.

File: test/suite/test_rollback_to_stable27.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+from test_rollback_to_stable01 import test_rollback_to_stable_base
+from wiredtiger import stat, Modify, WT_NOTFOUND
+from wtdataset import SimpleDataSet
+from wtscenario import make_scenarios
+
+# test_rollback_to_stable27.py
+#
+# Test mixing timestamped and non-timestamped updates on the same VLCS RLE cell.
+class test_rollback_to_stable28(test_rollback_to_stable_base):
```
[the remark this answers is not in the document]
Comment (sauclovian-wt) [reply test/suite/test_rollback_to_stable27.py]: Whoops. (Originally 27 was the larger test I wrote so this was 28; I renumbered, but evidently missed some bits)

Fixed.

File: test/suite/test_rollback_to_stable27.py (hunk 1 above)
Comment (sauclovian-wt) [reply test/suite/test_rollback_to_stable27.py]: Fixed.

**Comment by sauclovian-wt** [comment]
Thanks!

**Pull request merged**